src/command.rs

```diff
@@ -271,226 +305,351 @@ pub fn parse(input: &str) -> Option<Vec<Command>> {
 
     let mut commands = vec![];
     for command_input in command_inputs {
-        let components: Vec<_> = command_input.trim().split(' ').collect();
+        lazy_static! {
+            // https://docs.rs/regex/latest/regex/#example-avoid-compiling-the-same-regex-in-a-loop
+            static ref CONTINUOUS_SPACE_MATCHER: Regex = Regex::new(r" +").unwrap();
+        }
+        let command_sanitised = CONTINUOUS_SPACE_MATCHER.replace_all(&command_input, " ");
+        let components: Vec<_> = command_sanitised.trim().split(' ').collect();
```

Comment:
Maybe `str.split_whitespace()` is a simpler alternative we can use?

https://doc.rust-lang.org/std/primitive.str.html#method.split_whitespace

Reply:
Oh bollocks. I knew about this function for some time now, but never understood its point nor have read its documentation. It is better indeed. Thanks for showing me the way.